src/OpenSSL/SSL.py

```diff
@@ -1637,35 +1637,46 @@ def server_random(self):
 
         :return: A string representing the state
         """
-        if self._ssl.session == _ffi.NULL:
+        session = _lib.SSL_get_session(self._ssl)
+        if session == _ffi.NULL:
             return None
-        return _ffi.buffer(
-            self._ssl.s3.server_random,
-            _lib.SSL3_RANDOM_SIZE)[:]
+        length = _lib.SSL_get_server_random(self._ssl, _ffi.NULL, 0)
+        assert length > 0
```

Comment:
I have mixed feelings about adding asserts to code since they get optimized out.  Didn’t we just remove all kinds of asserts from cryptography?